connect/runtime/src/main/java/org/apache/kafka/connect/util/KafkaBasedLog.java

```diff
@@ -255,20 +255,15 @@ private void readToLogEnd() {
         }
 
         Map<TopicPartition, Long> endOffsets = new HashMap<>();
-        try {
-            poll(0);
```

Comment:
Removing `poll(0)` won't work here -- it was included for a reason. The `seekToEnd` calls don't reset the consumer position immediately, so you need to invoke `poll` before trying to get the updated positions.

It sounds like in some cases you were seeing that the problem was that the `poll()` was returning data unexpectedly. It does sound like this could be a potential problem -- it looks like this code was assuming that the seek would ensure that no data was delivered, but I think it may be possible that if you seek to the end and are already *at* the end, despite internally setting the position to `null` temporarily, you could receive a FetchResponse at just the right time such that the position is updated back to the same value and the data is delivered. We may need some sort of flag during offset reset to force the consumer to ignore any returned data temporarily.

However, from what I can see in the original bug report stack trace, the error occurred on the subsequent `poll()` call. Was that because the error just doesn't show up until later or is this really addressing a different bug? If it's a separate bug, it's worth fixing but it shouldn't be marked to resolve the 2944 unless we have a clear explanation of how it resulting in the behavior reported in 2944.

Of course, all of this would be moot if https://issues.apache.org/jira/browse/KAFKA-2500 were resolved since we wouldn't have to use this hacky approach to getting the log end offset.